src/arbitrary/sparseArray.ts

```diff
@@ -62,16 +70,26 @@ function arrayFromItems<T>(length: number, indexesAndValues: [number, T][]) {
  */
 export function sparseArray<T>(arb: Arbitrary<T>, constraints: SparseArrayConstraints = {}): Arbitrary<T[]> {
   const {
+    size,
     minNumElements = 0,
-    // TODO(size) - While from a generate point of view we want some limits using maxLengthFromMinLength
-    // makes us unable to shrink user inputs we could have potentially generated as we restrict our array
-    // to a smaller one (for generate reasons). Ideally maxLength should only be specified when there is
-    // a real need for them. For any other case, we should try not to pass them.
-    maxNumElements = maxLengthFromMinLength(minNumElements, DefaultSize),
-    maxLength = maxLengthFromMinLength(maxNumElements, DefaultSize),
+    maxNumElements = MaxLengthUpperBound,
+    maxLength = MaxLengthUpperBound,
     noTrailingHole,
   } = constraints;
 
+  const maxGeneratedNumElements = maxGeneratedLengthFromSizeForArbitrary(
+    size,
+    minNumElements,
+    maxNumElements,
```

Comment:
There is possibly something to check regarding maxLength and maxNumElements if one has been specified (by the user) but not the other

Reply:
Addressed with https://github.com/dubzzz/fast-check/pull/2690/files#r796032170